src/Storages/PartitionStrategy.h

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#include <Interpreters/ExpressionActions.h>
+#include <Parsers/IAST_fwd.h>
+
+#include <Processors/Chunk.h>
+
+namespace DB
+{
+
+struct PartitionStrategy
+{
+    struct PartitionExpressionActionsAndColumnName
+    {
+        ExpressionActionsPtr actions;
+        std::string column_name;
+    };
+
+    static std::unordered_map<std::string, bool> partition_strategy_to_wildcard_acceptance;
```

Comment:
A comment please. It might not be clear immediately what `wildcard_acceptance` is.